src/runtimes/node/bundlers/types.ts

```diff
@@ -5,7 +5,25 @@ import type { FeatureFlag, FeatureFlags } from '../../../feature_flags.js'
 import type { FunctionSource } from '../../../function.js'
 import type { ModuleFormat } from '../utils/module_format.js'
 
-export type NodeBundlerName = 'esbuild' | 'esbuild_zisi' | 'nft' | 'zisi'
+export const enum NodeBundlerType {
+  ESBUILD = 'esbuild',
+  ESBUILD_ZISI = 'esbuild_zisi',
+  NFT = 'nft',
+  ZISI = 'zisi',
+  SKIP = 'skip',
+}
+
+export interface NoOptionsConfig<
```

Comment:
Something like:
```ts
interface FunctionConfig<Bundler extends NodeBundlerType = NodeBundlerType> {
  nodeBundler?: Bundler,
  nodeBundlerOptions: Bundler extends NodeBundlerType.SKIP ? SkipConfig: never 
}
```

Problem here is that nodeBundlerOptions would be either mandatory or not mandatory for all cases.
This could be worked around with:

```ts
interface BaseFunctionConfig {
...
}

type FCWithoutOptions = BaseFunctionConfig & { nodeBundler: Exclude<NodeBundlerType, NodeBundlerType.SKIP> }
type FCWithSkipOptions = BaseFunctionConfig & { nodeBundler: NodeBundlerType.SKIP, nodeBundlerOptions: SkipOptions }

type FunctionConfig = FCWithoutOptions | FCWithSkipOptions
```

I personally feel this is similar complicated, just in a different location. And it is less extensible in case one of the other Bundlers starts to have options.

What I did now is completely move the logic for the nodeBundler field into the bundlers/types.ts file, so it is in one place and also tried to simplify it a little bit.

Reply:
> Problem here is that nodeBundlerOptions would be either mandatory or not mandatory for all cases.

What if it's never mandatory? The worst thing that could happen is that someone sets `nodeBundler: "custom"` and doesn't provide a `nodeBundlerOptions` object so we don't include any files in the bundle? Given that this will be used for very specific use cases and advanced integrations, I think that would be an okay tradeoff if it means simplifying the implementation.

I can also see an argument for having rigid types, so I'm happy to defer to you.